python_ta/transforms/type_inference_visitor.py

```diff
@@ -378,18 +376,9 @@ def visit_for(self, node):
     # Comprehensions
     ##############################################################################
     def visit_comprehension(self, node: astroid.Comprehension) -> None:
-        # TODO: refactor code duplication between this and visit_for.
-        iter_type = self._handle_call(node, '__iter__', node.iter.inf_type.getValue()).getValue()
-        contained_type = iter_type.__args__[0]
-        if isinstance(node.target, astroid.AssignName):
-            target_type = self.lookup_type(node.target, node.target.name)
-        else:
-            # TODO: check whether the following assumption is valid
-            # Assume that node.target is a tuple.
-            target_type = Tuple[tuple(self.lookup_type(subtarget, subtarget.name) for subtarget in node.target.elts)]
-
-        self.type_constraints.unify(contained_type, target_type)
-        node.inf_type = TypeInfo(NoType)
+        for if_stmt in node.ifs:
+            self.type_constraints.unify(node.if_stmt.inf_type.getValue(), bool)
```

Comment:
Looking at the visitor for IfExpr, I just realized that this is unnecessary since `if` conditions can be any type. Shall I remove these two lines?

Reply:
@simeonkr yeah, you can remove these.